packages/react-router-dev/config/config.ts

```diff
@@ -404,12 +402,11 @@ async function resolveConfig({
           return null;
         }
 
-        let configPreset: ReactRouterConfig = omit(
+        return omit(
           await preset.reactRouterConfig({ reactRouterUserConfig }),
+          // @ts-expect-error
```

Comment:
what error is this suppressing? 👀

Reply:
> because a `ConfigPreset` doesn't have a `presets` key, right?

Indeed.

> you can cast `as never`

Yeah I don't like casting, hence the `expect-error`, but I guess @brophdawg11 can just tell me the preference of the team and I'll update to whatever is preferred

Reply:
The expect-error is nice because we'll get a type error if/when we fix types for `ConfigPreset`, at which point we can clean this up. Whereas with a cast, that cast will likely live in the codebase forever.

Reply:
@pcattori That's indeed (one of) the reason(s) why I don't like casting.